nucliadb_sdk/nucliadb_sdk/v2/sdk.py

```diff
@@ -0,0 +1,290 @@
+                else:
+                    data = content.json()
+            else:
+                # pull properties out of kwargs now
+                content_data = {}
+                for key in list(kwargs.keys()):
+                    if key in request_type.__fields__:
+                        content_data[key] = kwargs.pop(key)
+                data = request_type.parse_obj(content_data).json()
+
+        query_params = kwargs.pop("query_params", None)
+        if len(kwargs) > 0:
+            raise TypeError(f"Invalid arguments provided: {kwargs}")
+
+        resp = self._request(path, method, data=data, query_params=query_params)
+
+        if response_type is not None:
+            if isinstance(response_type, type) and issubclass(response_type, BaseModel):
+                return response_type.parse_raw(resp.content)  # type: ignore
+            else:
+                return response_type(resp)  # type: ignore
+        else:
+            return resp.content
+
+    return _func
+
+
+class NucliaSDK:
+    """
+    Example usage:
+
+    from nucliadb_sdk.v2.sdk import *
+    sdk = NucliaSDK(region=Region.EUROPE1, api_key="api-key")
+    sdk.list_resources(kbid='70a2530a-5863-41ec-b42b-bfe795bef2eb')
+    """
+
+    def __init__(
+        self,
+        *,
+        region: Region = Region.EUROPE1,
+        api_key: Optional[str] = None,
+        url: Optional[str] = None,
+        headers: Optional[dict[str, str]] = None,
+    ):
+        self.region = region
+        self.api_key = api_key
+        headers = headers or {}
+        if region == Region.ON_PREM:
+            if url is None:
+                raise ValueError("url must be provided for on-prem")
+            self.base_url = url.rstrip("/")
+            # By default, on prem should utilize all headers available
+            # For custom auth schemes, the user will need to provide custom
+            # auth headers
+            headers["X-NUCLIADB-ROLES"] = "MANAGER;WRITER;READER"
+        else:
+            if api_key is None:
+                raise ValueError("api_key must be provided for cloud sdk usage")
+            self.base_url = f"https://{region.value}.nuclia.cloud/api"
+            headers["X-STF-SERVICEACCOUNT"] = f"Bearer {api_key}"
+
+        self.session = httpx.Client(headers=headers, base_url=self.base_url)
+
+    def _request(
+        self,
+        path,
+        method: str,
+        data: Optional[Union[str, bytes]] = None,
+        query_params: Optional[dict[str, str]] = None,
+    ):
+        url = f"{self.base_url}{path}"
+        opts: dict[str, Any] = {}
+        if data is not None:
+            opts["data"] = data
+        if query_params is not None:
+            opts["params"] = query_params
+        response: httpx.Response = getattr(self.session, method.lower())(url, **opts)
+
+        if response.status_code < 300:
+            return response
+        elif response.status_code in (401, 403):
```

Comment:
We are using 402 code for billing-related limit errors. Look at https://github.com/nuclia/nucliadb/blob/main/nucliadb/nucliadb/ingest/processing.py#L290

or 

https://github.com/nuclia/nucliadb/blob/main/nucliadb/nucliadb/search/predict.py#L182-L184